src/main/java/com/google/devtools/build/lib/exec/StandaloneTestStrategy.java

```diff
@@ -463,6 +463,7 @@ private static Spawn createCoveragePostProcessingSpawn(
         action.getLcovMergerRunfilesSupplier(),
         /* filesetMappings= */ ImmutableMap.of(),
         /* inputs= */ NestedSetBuilder.<ActionInput>compileOrder()
+            .addAll(ImmutableList.copyOf(action.getInputs().toList()))
```

Comment:
>  Is there a particular reason you're already using it?

It is the easiest fix for actually getting remote execution to return non-empty `coverage.dat` files for `cc_test` unless you have another easier method? 

Hmm I thought I'd tried `addAll(action.getInputs())` but quickly testing this again, it seems to also compile. I'll do some further testing again today.

Reply:
Thanks a lot for this change. Does it work now and I can merge? Also if it's not too much to ask, do you mind adding a test similar to this [one](https://cs.opensource.google/bazel/bazel/+/master:src/test/shell/bazel/remote/remote_execution_test.sh;l=2346) that includes what was failing for you?

Reply:
Is there a way for me to see the URLs you send as I always get, "permission denied".

Yes the current fix works for RE and Bazel Coverage. I'll take a look at creating some unittests tomorrow and come back at you with any questions if I have any.

Reply:
@oquenchil - Done! Can we look to getting this merged now?

Reply:
@oquenchil - I've added a test. There are two minor difficulties with this. Firstly the coverage report changes depending on the toolchain. Originally I had it so that it checked against an actual report but have no just updated it to check that it is non-zero.

Now it fails because of centos. Are you able to see what might have happened here?

Reply:
@ulfjack @oquenchil , I've found out why it fails for centos. It's because the `gcov` version is too old:

```
gcov (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44)
$ gcov --help
Usage: gcov [OPTION]... SOURCE|OBJ...

Print code coverage information.

  -h, --help                      Print this help, then exit
  -v, --version                   Print version number, then exit
  -a, --all-blocks                Show information for every basic block
  -b, --branch-probabilities      Include branch probabilities in output
  -c, --branch-counts             Given counts of branches taken
                                    rather than percentages
  -n, --no-output                 Do not create an output file
  -l, --long-file-names           Use long output file names for included
                                    source files
  -f, --function-summaries        Output summaries for each function
  -o, --object-directory DIR|FILE Search for object files in DIR or called FILE
  -s, --source-prefix DIR         Source prefix to elide
  -r, --relative-only             Only show data for relative sources
  -p, --preserve-paths            Preserve all pathname components
  -u, --unconditional-branches    Show unconditional branch counts too
  -d, --display-progress          Display progress information
```

Reply:
@ulfjack @oquenchil , I've figured out why it fails for centos. It's because the `gcov` version is too old and it doesn't include the option for outputting to an intermediate text format:
```
gcov (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44)
$ gcov --help
Usage: gcov [OPTION]... SOURCE|OBJ...

Print code coverage information.

  -h, --help                      Print this help, then exit
  -v, --version                   Print version number, then exit
  -a, --all-blocks                Show information for every basic block
  -b, --branch-probabilities      Include branch probabilities in output
  -c, --branch-counts             Given counts of branches taken
                                    rather than percentages
  -n, --no-output                 Do not create an output file
  -l, --long-file-names           Use long output file names for included
                                    source files
  -f, --function-summaries        Output summaries for each function
  -o, --object-directory DIR|FILE Search for object files in DIR or called FILE
  -s, --source-prefix DIR         Source prefix to elide
  -r, --relative-only             Only show data for relative sources
  -p, --preserve-paths            Preserve all pathname components
  -u, --unconditional-branches    Show unconditional branch counts too
  -d, --display-progress          Display progress information
```
Which is required here:
https://github.com/bazelbuild/bazel/blob/e6f1066deef066df82e3e2771ef521c03c252f01/tools/test/collect_cc_coverage.sh#L147

You can see the error in the logs if you search `_coverage/gcov: invalid option`:
https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/16de9e2a-0d65-45a6-b06d-5007aadee2d3/src/test/shell/bazel/remote/remote_execution_test/attempt_1.log

What do you advise? Can we simply skip these coverage tests for centos or do you want another workaround?

Reply:
Thanks a lot for this! This is awesome.

You can definitely add a check to skip the test if you detect it would fail because the gcov version is too old.

Reply:
@oquenchil - ok done! I think the tests are currently failing I think because of a bazel side infra problem. Does that all look ok to you now and can we get this lined up to be merged?

Reply:
@oquenchil - can you check why it's failing? I currently `return` if it can't grep for the `gcov` flag. However that seems to cause a strange error in Bazel's CI:

https://storage.googleapis.com/bazel-untrusted-buildkite-artifacts/4855b77c-41bc-47ca-935a-5f424538d1de/src/test/shell/bazel/remote/remote_execution_test/attempt_1.log

Reply:
This is because of something known with our shell testing. If you run grep and it fails it will fail the test immediately without running the next line. We avoid that by having something else behind like `grep something || fail "test failed"`. In this case, you don't want fail the test just return, so maybe you can do `grep something || return`. If that doesn't work, then perhaps you can store the output of gcov --help in a variable and check for the presence of "-i" in the conditional in some other way with bash.

Reply:
Done, that worked!

Can we get this merged @oquenchil ?